expo/notifications.go

```diff
@@ -0,0 +1,198 @@
+package expo
+
+// Adapted from https://github.com/oliveroneill/exponent-server-sdk-golang.
+// We aren't using that package directly because it does not support the
+// use of Expo Access Tokens or handling the Receipt ID. There are already
+// PR's open for these issues, but it seems like the package is no longer
+// maintained.
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+)
+
+type PushMessage struct {
+	To    string `json:"to"`
+	Title string `json:"title,omitempty"`
+	Body  string `json:"body"`
+}
+
+type Response struct {
+	Data   []PushResponse      `json:"data"`
+	Errors []map[string]string `json:"errors"`
+}
+
+type PushResponse struct {
+	PushMessage PushMessage
+	ID          string            `json:"id"` // receipt
+	Status      string            `json:"status"`
+	Message     string            `json:"message"`
+	Details     map[string]string `json:"details"`
+}
+
+func (r *PushResponse) isSuccess() bool {
+	return r.Status == SuccessStatus
+}
+
+func (r *PushResponse) ValidateResponse() error {
+	if r.isSuccess() {
+		return nil
+	}
+	err := &PushResponseError{
+		Response: r,
+	}
+	// Handle specific errors if we have information
+	if r.Details != nil {
+		e := r.Details["error"]
+		if e == ErrorDeviceNotRegistered {
```

Comment:
You can write this as:

```
switch r.Details["error"] {
case ErrorDeviceNotRegistered:
  ...
case ErrorMessageTooBig:
  ...
...
}
```

Reply:
I'm using the official expo package now for this functionality.